### PR TITLE
Added Findcppnetlib.cmake file.

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

**Comment by eile** [comment]
Where does the file come from?

At least _QUIETLY should be respected, and the messages should follow the standard convention. See for example https://github.com/Eyescale/CMake/blob/master/FindDNSSD.cmake.

File: Findcppnetlib.cmake [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+#
+
+# Assume not found.
+SET(CPPNETLIB_FOUND FALSE)
+
+# PATH ________________________________________________________________________
+
+if (CPPNETLIB_PATH)
+    # Set by user...
+else (CPPNETLIB_PATH)
```
Comment (eile) [inline Findcppnetlib.cmake]: if(NOT CPPNETLIB_PATH)

File: Findcppnetlib.cmake [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    find_path(CPPNETLIB_PATH include/boost/network.hpp
+	HINTS ${CPP_NETLIB_ROOT} $ENV{CPP_NETLIB_ROOT}
+        /usr/local/
+        /usr/
+    )
+endif (CPPNETLIB_PATH)
```
Comment (eile) [inline Findcppnetlib.cmake]: use endif(), not endif([repeatedifcond])

File: Findcppnetlib.cmake [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        set(_cppnetlib_output 1)
+    endif()
+endif (CPPNETLIB_FIND_REQUIRED)
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(CPPNETLIB DEFAULT_MSG CPPNETLIB_LIBRARIES CPPNETLIB_INCLUDE_DIR)
+
+set(CPPNETLIB_INCLUDE_DIRS ${CPPNETLIB_INCLUDE_DIR})
+
+if (NOT CPPNETLIB_FOUND)
```
Comment (eile) [inline Findcppnetlib.cmake]: AND _output

File: Findcppnetlib.cmake [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    message( ${_cppnetlib_output_type} "Could not find cpp-netlib")
+endif (NOT CPPNETLIB_FOUND)
```
Comment (eile) [inline Findcppnetlib.cmake]: rm three lines - is already done in FindPackageHandleStandardArgs.

Comment (eile) [inline Findcppnetlib.cmake]: This is still open.

File: Findcppnetlib.cmake [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+#
+
+# Assume not found.
+SET(CPPNETLIB_FOUND FALSE)
+
+# PATH ________________________________________________________________________
+
+if (NOT CPPNETLIB_PATH)
+    find_path(CPPNETLIB_PATH include/boost/network.hpp
+	HINTS ${CPP_NETLIB_ROOT} $ENV{CPP_NETLIB_ROOT}
```
Comment (tribal-tec) [inline Findcppnetlib.cmake]: Inconsistency in naming. CMake variable as CPPNETLIB, but the env variable has an underscore in between. I suggest to use the non-underscore one for all variables. Also document the env variable in the comment block above accordingly.

File: Findcppnetlib.cmake [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+# CPPNETLIB_LIBRARIES - The cppnetlib libraries
+#==============================================================================
+#
+
+# Assume not found.
+SET(CPPNETLIB_FOUND FALSE)
+
+# PATH ________________________________________________________________________
+
+if (NOT CPPNETLIB_PATH)
```
Comment (tribal-tec) [inline Findcppnetlib.cmake]: please remove the space between if and the braces in all places of this file.

File: Findcppnetlib.cmake [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+# HEADERS AND DYNAMIC LIBRARIES_________________________________________________
+
+if (CPPNETLIB_PATH)
+    set (CPPNETLIB_INCLUDE_DIR ${CPPNETLIB_PATH}/include)
+    mark_as_advanced (CPPNETLIB_INCLUDE_DIR)
+
+    find_library(CPPNETLIB_LIBRARY1 NAMES cppnetlib-client-connections
+	HINTS ${CPPNETLIB_ROOT} $ENV{CPPNETLIB_ROOT}
+        PATHS ${CPPNETLIB_PATH}/lib64
```
Comment (tribal-tec) [inline Findcppnetlib.cmake]: check for /lib as well?

File: Findcppnetlib.cmake [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+# FOUND _______________________________________________________________________
+if (CPPNETLIB_FIND_REQUIRED)
+    set(_cppnetlib_output_type FATAL_ERROR)
+else()
+    set(_cppnetlib_output_type STATUS)
+endif ()
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(CPPNETLIB DEFAULT_MSG CPPNETLIB_LIBRARIES CPPNETLIB_INCLUDE_DIR)
```
Comment (tribal-tec) [inline Findcppnetlib.cmake]: no need for UPPER_CASE function name. Lower-case also works and looks less aggressive :)

**Comment by eile** [comment]
+1

**Pull request merged**